### PR TITLE
Added HTTP_SKIP_CERT_VALIDATION and http implementation to http util

### PR DESCRIPTION
**Pull request by MaartendeKruijf**

### CODE REVIEW

**Review by RabbITCybErSeC — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -45,23 +45,28 @@ jobs:
...
+      - name: Start docker containers for test
+        run: docker-compose -f "docker/testing/httpbin-test/docker-compose.yml" up -d --build
       - name: Run tests
         run: |
-          apt update
-          apt install openssh-server -y
-          useradd sshtest
-          echo "sshtest:pdKY77qNxpI5MAizirtjCVOcm0KFKIs" | chpasswd
-          service ssh start
+          sudo apt update
```
Comment (MaartendeKruijf) [inline .github/workflows/ci.yml:66]: Why does it need sudo ?

File: test/unittest/utils/http/http_test.go [hunk 2]
```diff
@@ -34,6 +34,61 @@ type httpBinResponseBody struct {
...
+		Headers: map[string][]string{"accept": {"application/json"}},
+	}
+	httpOptions := http.HttpOptions{
+		Command: &command,
+		Target:  &target,
+	}
+
+	response, err := httpRequest.Request(httpOptions)
+	t.Log(string(response))
+	if err == nil {
```
Comment (MaartendeKruijf) [inline test/unittest/utils/http/http_test.go:87]: use assert here

File: test/unittest/utils/http/http_test.go [hunk 3]
```diff
@@ -34,6 +34,61 @@ type httpBinResponseBody struct {
...
+		Headers: map[string][]string{"accept": {"application/json"}},
+	}
+	httpOptions := http.HttpOptions{
+		Command: &command,
+		Target:  &target,
+	}
+	httpRequest.SkipCertificateValidation(true)
+	response, err := httpRequest.Request(httpOptions)
+	t.Log(string(response))
+	if err != nil {
```
Comment (MaartendeKruijf) [inline test/unittest/utils/http/http_test.go:58]: assert err equals nil

File: test/unittest/utils/http/http_test.go [hunk 4]
```diff
@@ -34,6 +34,61 @@ type httpBinResponseBody struct {
 }
 
 // Test general http options, we do not check responses body, as these are variable for the general connection tests
+
+func TestInsecureHTTPConnection(t *testing.T) {
```
Comment (MaartendeKruijf) [inline test/unittest/utils/http/http_test.go:38]: Move test to integration tests

File: .github/workflows/ci.yml (hunk 1 above)
Comment (RabbITCybErSeC) [reply .github/workflows/ci.yml:66]: because it is not root anymore

Comment (RabbITCybErSeC) [reply .github/workflows/ci.yml:66]: I removed the container section that required root

**Review by RabbITCybErSeC — APPROVED** [review]

**Pull request merged**